core/kernel/interrupt.c

```diff
@@ -37,51 +41,205 @@
  */
 
 static struct itr_chip *itr_chip;
-static SLIST_HEAD(, itr_handler) handlers = SLIST_HEAD_INITIALIZER(handlers);
+static struct itr_desc *itr_descs[256] = {0};
```

Comment:
With this array you waste the larger part of 1 kB. Also there's now a fixed upper id of an interrupt.
I'd rather keep the old dynamic linked list, I don't expect that we'll have that many interrupt handlers in secure world that we'll have a problem of scaling.